homeassistant/components/roborock/image.py

```diff
@@ -66,17 +66,26 @@ def __init__(
         )
         self._attr_image_last_updated = dt_util.utcnow()
         self.map_flag = map_flag
-        self.cached_map = self._create_image(starting_map)
+        try:
+            self.cached_map = self._create_image(starting_map)
+        except HomeAssistantError:
+            # If we failed to update the image on init, we set cached_map to empty bytes so that we are unavailable and can try again later.
+            self.cached_map = b""
         self._attr_entity_category = EntityCategory.DIAGNOSTIC
 
+    @property
+    def available(self):
+        """Determines if the entity is available."""
+        return self.cached_map != b""
+
     @property
     def is_selected(self) -> bool:
         """Return if this map is the currently selected map."""
         return self.map_flag == self.coordinator.current_map
 
     def is_map_valid(self) -> bool:
-        """Update this map if it is the current active map, and the vacuum is cleaning."""
-        return (
+        """Update this map if it is the current active map, and the vacuum is cleaning or if it has never been set at all."""
```

Comment:
Please follow PEP8 for docstrings. Max 72 characters per line and the first line is a header and should be a single sentence. An optional body can follow after a blank line.